src/KMonad/Model/Dispatch.hs

```diff
@@ -58,20 +60,21 @@ import qualified RIO.Text as T
 data Dispatch = Dispatch
   { _eventSrc :: IO KeyEvent            -- ^ How to read 1 event
   , _readProc :: TMVar (Async KeyEvent) -- ^ Store for reading process
-  , _rerunBuf :: TVar (Seq KeyEvent)    -- ^ Buffer for rerunning events
+  , _rerunBuf :: TVar (Seq WrappedEvent)    -- ^ Buffer for rerunning events
+  , _injectTmr :: TMVar Unique -- ^ Used to signal timeouts
   }
 makeLenses ''Dispatch
 
 -- | Create a new 'Dispatch' environment
-mkDispatch' :: MonadUnliftIO m => m KeyEvent -> m Dispatch
-mkDispatch' s = withRunInIO $ \u -> do
+mkDispatch' :: MonadUnliftIO m => m KeyEvent -> TMVar Unique -> m Dispatch
+mkDispatch' s itr = withRunInIO $ \u -> do
```

Comment:
As mentioned above I would change the order of parameters like this and also for `mkDispatch`, `mkHooks`, ... (and maybe `Dispatch`, ...)
```suggestion
mkDispatch' :: MonadUnliftIO m => TMVar Unique -> m KeyEvent -> m Dispatch
mkDispatch' itr s = withRunInIO $ \u -> do
```